src/main/java/de/tum/in/www1/artemis/service/connectors/Lti10Service.java

```diff
@@ -0,0 +1,287 @@
+
+    private final Logger log = LoggerFactory.getLogger(Lti10Service.class);
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    private final LtiService ltiService;
+
+    private final LtiOutcomeUrlRepository ltiOutcomeUrlRepository;
+
+    private final ResultRepository resultRepository;
+
+    private final HttpClient client;
+
+    public Lti10Service(UserRepository userRepository, LtiOutcomeUrlRepository ltiOutcomeUrlRepository, ResultRepository resultRepository, CourseRepository courseRepository,
+            LtiService ltiService) {
+        this.userRepository = userRepository;
+        this.ltiOutcomeUrlRepository = ltiOutcomeUrlRepository;
+        this.resultRepository = resultRepository;
+        this.courseRepository = courseRepository;
+        this.ltiService = ltiService;
+        this.client = HttpClientBuilder.create().build();
+    }
+
+    /**
+     * Checks if an LTI request is correctly signed via OAuth with the secret
+     *
+     * @param request The request to check
+     * @param onlineCourseConfiguration The configuration containing the secret used to verify the request
+     * @return null if the request is valid, otherwise an error message which indicates the reason why the verification failed
+     */
+    public String verifyRequest(HttpServletRequest request, OnlineCourseConfiguration onlineCourseConfiguration) {
+        if (onlineCourseConfiguration == null) {
+            String message = "verifyRequest for LTI is not supported for this course";
+            log.warn(message);
+            return message;
+        }
+
+        LtiVerifier ltiVerifier = new LtiOauthVerifier();
+        try {
+            LtiVerificationResult ltiResult = ltiVerifier.verify(request, onlineCourseConfiguration.getLtiSecret());
+            if (!ltiResult.getSuccess()) {
+                String requestString = httpServletRequestToString(request);
+                final var message = "LTI signature verification failed with message: " + ltiResult.getMessage() + "; error: " + ltiResult.getError() + ", launch result: "
+                        + ltiResult.getLtiLaunchResult();
+                log.error(message);
+                log.error("Request: {}", requestString);
+                return message;
+            }
+        }
+        catch (LtiVerificationException e) {
+            log.error("Lti signature verification failed. ", e);
+            return "Lti signature verification failed; " + e.getMessage();
+        }
+        // this is the success case
+        log.info("LTI Oauth Request Verification successful");
+        return null;
+    }
+
+    /**
+     * helper method to print a request with all its elements
+     *
+     * @param request the http request
+     * @return a string with all debug information
+     */
+    private String httpServletRequestToString(HttpServletRequest request) {
+        StringBuilder requestString = new StringBuilder();
```

Comment:
I think there should be java classes for those use cases like https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpRequest.Builder.html instead of building it "by hand"

Reply:
Same as above